packages/babel-core/src/transformation/file/index.js

```diff
@@ -288,14 +288,15 @@ export default class File extends Store {
 
     let generator = this.get("helperGenerator");
     let runtime   = this.get("helpersNamespace");
+    let res = null;
     if (generator) {
-      let res = generator(name);
-      if (res) return res;
+      res = generator(name);
+      if (res && t.isIdentifier(res)) return res;
```

Comment:
I don't totally follow the logic here, could you clarify for me, what's the result of changing this check?